### ISSUE TITLE
Error: failed to close reader when asking treefmt to format files outside of the tree root if they're specified as a relative path

### ISSUE THREAD

**Issue by jfly** [numtide/treefmt#444]
Here's what happens if I'm in some project and use a relative path to ask treefmt to format a file outside of the project:

```
proj/ $ treefmt --no-cache ../out-of-tree.py
traversed 0 files
emitted 0 files for processing
formatted 0 files (0 changed) in 2ms
Error: failed to close reader: path '/tmp/tmp.V5f56DPrFE/out-of-tree.py' is outside of the root '/tmp/tmp.V5f56DPrFE/proj'
```

This is different than nice "not found within the tree root" error message we normally get:

```
proj/ $ treefmt --no-cache /out-of-tree.py
Error: path /out-of-tree.py not found within the tree root /tmp/tmp.V5f56DPrFE/proj
```

**To Reproduce**

Here's the setup for the above commands:

```
$ cd "$(mktemp -d)"
$ touch out-of-tree.py
$ mkdir proj
$ cd proj
$ git init

$ nix shell github:numtide/treefmt nixpkgs#black
$ treefmt --version
treefmt v659aa0f
$ cat treefmt.toml   # create this file
[formatter.python]
command = "black"
includes = ["*.py"]
```

**Expected behavior**

I acknowledge this is a pretty nitty cosmetic issue. That said, I would expect treefmt to give the same error message for files outside of the tree root.

**System information**

See above. I'm using 659aa0f.

**Comment by jfly** [comment]
Fix out for review here: https://github.com/numtide/treefmt/pull/445/

**Comment by brianmcgee** [comment]
Fixed with #445
